types/Config.js

```diff
@@ -190,7 +190,7 @@ export type InitialOptions = {
 export type SnapshotUpdateState = 'all' | 'new' | 'none';
 
 export type GlobalConfig = {|
-  bail: boolean,
+  bail: number,
```

Comment:
bail is not normalized to be a number

Reply:
Bail does get normalized: https://github.com/facebook/jest/pull/7335/files#diff-7121e12d406ff9a3060a13ee808632de

Reply:
Hm, why would `bail` fall into `testPathPattern`, do you have an example? Pushing to `argv._` is definitely something that we'd like to avoid.

Reply:
`jest --bail someTestPattern` will push that argument into the `bail` argv option and it won't exist in `argv._` which is where it goes otherwise.

Reply:
I see. So, since we're breaking quite some stuff in v24, maybe let's just make it only accept numbers? wdyt @SimenB? If not, I think the proposed solution makes sense, but needs to be guarded in a thorough comment, as it mutates `argv`.

Reply:
Any suggestions on how to handle the flow error?  `argv._` is typed as `Array<string>` and `bail` is either `number|boolean` so it gets angry in this case.  I'm not sure what the recommended way of handling that flow error would be in this case?

Reply:
maybe instead of `isNan` you can check for `typeof string`?